app/modules/asset.rb

```diff
@@ -0,0 +1,148 @@
+class Asset
```

Comment:
I don’t think `Asset` is a great name, it doesn’t describe well the responsibilities of the object. The object looks like it is retrieving the analytics data for a specific time window. This data is then used to generate reports. Now based on my assumptions, I think a name like `AnalyticsWindowReport` would be more appropriate (or anything else, as long as it covers the responsibilities nicely).